lib/straight.gi

```diff
@@ -2477,7 +2477,6 @@ InstallGlobalFunction( IntermediateResultsOfSLPWithoutOverwriteInner,
                 AddSet(needed,line[j]);
```

Comment:
This looks weird, as we add things to 'needed' during this loop, which could change the last element of 'needed'?

Reply:
Another good point, but this one I thought about before, *phew* :-). But here is why I think this is correct: we are looking at a straight line program. The definition for the value in slot $i$ may depend on other slots, but *only* slots preceding $i$. Note that `line` here is the "program" used o define slot $i$, and `line[j]` is the id of a slot used to define slot $i$.

Reply:
I believe you, just because we are editing something that doesn't get that much analysis, you could chuck an `Assert(2, line[j] < i)` (if that's right?), just in case some combination of errors is cancelling out (hopefully a mostly unused assert won't be a notable performance hit, they are reasonably quick nowadays), but I do believe that reasoning is correct.

Reply:
I've added the `Assert`, it's a good idea